build-tools/packages/build-cli/src/commands/setReleaseTagForPublishing.ts

```diff
@@ -0,0 +1,104 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Flags } from "@oclif/core";
+import { Package, updatePackageJsonFile } from "@fluidframework/build-tools";
+import path from "node:path";
+import { PackageCommand } from "../BasePackageCommand";
+import * as fs from "fs";
+
+/**
+ * A type defining the four release types
+ */
+type UpdatePackageJsonType = "alpha" | "beta" | "public";
+
+/**
+ * A typeguard to check if a string is a {@link UpdatePackageJsonEnum}.
+ */
+function isUpdatePackageJsonTypes(type: string): type is UpdatePackageJsonType {
+	return type === "alpha" || type === "beta" || type === "public";
+}
+
+export default class SetReleaseTagPublishingCommand extends PackageCommand<
+	typeof SetReleaseTagPublishingCommand
+> {
+	static readonly description =
+		"Updates the types in package.json based on the release flag. This command is intended for use in publishing pipelines and may not be intended for regular developer use.";
+
+	static flags = {
+		types: Flags.custom<UpdatePackageJsonType>({
```

Comment:
Defining an array with the values will let you provide an `options` property, which improves the help as well as auto-complete in the terminal.